site/content/blog/2020-02-10-maintaining-consistency-in-codebases-with-go-vet.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Maintaining consistency in codebases with Go Vet
+slug: maintaining-consistency-in-codebases-with-go-vet
+date: 2020-02-10
+categories:
+    - "go"
+author: Jesús Espino
+github: jespino
+community: jesus.espino
+---
+
+At Mattermost we are a bit obsessed with the quality of our code, and one of
+the things that we love is consistent and idiomatic Go. In that sense, go vet
+is a great tool, because it checks a lot of things for us (useless assignments,
+wrong number of Printf parameters, correct format in struct tags, etcetera) as
+part of our CI pipeline, helping us and our contributors to write better code.
```

Comment:
Maintaining success in a large open-source project is one of the key objectives of Mattermost. We have hundreds of contributors and we want to create a project that could serve as a model in the Go community. Having said that, following idiomatic Go principles is the thing that we care most while maintaining our code consistency. For this specific task, we utilized `go vet` and with this blog post, I would like to explain how we pushed the limits of this tool by extending it.